controller/cmd/destination/main.go

```diff
@@ -43,24 +43,27 @@ func main() {
 		log.Fatalf("Failed to listen on %s: %s", *addr, err)
 	}
 
+	global, err := config.Global(consts.MountPathGlobalConfig)
```

Comment:
I'd probably move this into the `destination` package into the `NewServer` function.

We don't need to and don't want to set a different cluster domain than what's set in the config so calling it here give us no benefit.

By moving it you reduce the amount of parameters required for `NewServer`.

---
Since `config.Global` is needed for the trust domain too, i'd also move that into the `destination` package and pass `*disableIdentity ` instead of `trustDomain` to `NewServer`

Reply:
I think I prefer to keep the config reading in the main and pass the specific values into `NewServer`.  This reduces coupling between the server implementation and the config, and makes the server easier to test.